### ISSUE TITLE
`GradleTest` fails alleging an issue with `jackson-bom` version `2.15.0-rc1`

### ISSUE THREAD

**Issue by cowtowncoder** [FasterXML/jackson-integration-tests#18]
So, after publishing `2.15.0-rc1`, `GradleTest` started failing with:

```
[ERROR] Failures: 
[ERROR]   GradleTest.testJacksonBomDependency:36->build:64 FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':checkMetadata'.
> Missing dependency to 'jackson-bom'. Gradle Metadata publishing is most likely broken:
    - com.fasterxml.jackson.core:jackson-core:2.15.0-rc1

```

At first I was wondering if that might be due to:

https://github.com/FasterXML/jackson-core/issues/955

but after upgrading (just to be sure), via #17 failure persists. Perhaps that's just due to oddities of handling of (Pre-)Release Candidates, version number. Or maybe I made some mistake in publishing. But would be good to figure out.

**Comment by cowtowncoder** [comment]
@jjohannes I was wondering if you had an idea of what is the problem here?

**Comment by jjohannes** [comment]
@cowtowncoder happy to check. The issue is valid. Which shows, at least, that this test is useful to some degree. Maybe we could print a more actionable message though.

Two things to check in such a case:
- Is the `.module` file present?
  It is: https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-core/2.15.0-rc1/jackson-core-2.15.0-rc1.module
- Does the `pom.xml` contain the comment that informs Gradle about the existence of the file?
  It does **not**: https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-core/2.15.0-rc1/jackson-core-2.15.0-rc1.pom

I am pretty sure it is the same problem as with `jackson-module-afterburner`. Here the shade plugin was added in `core` that rewrites the pom and removes the comment:
https://github.com/FasterXML/jackson-core/pull/843

We can do the same fix in the `core` pom.xml as here:
https://github.com/FasterXML/jackson-modules-base/pull/198/commits/6df8246f3805f2d3c0e35e0aeeb0e29ba27b34ad#diff-679477445fa9271ed890643e4b1e942bd43356a47c02f6e98e908d1538969787

**Comment by cowtowncoder** [comment]
Ah! Yes, that makes sense. Should have remembered...

**Comment by cowtowncoder** [comment]
Fixed via https://github.com/FasterXML/jackson-core/pull/966 I hope.
Will leave open until we can publish 2.15.0-rc2

**Comment by cowtowncoder** [comment]
2.15.0-rc2 was released, no longer failing, closing.

**Comment by cowtowncoder** [comment]
With Jackson 2.15.1 release (https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.15.1), getting closer to getting `GradleTest.testJacksonBomDependency` to pass -- mr-bean module the lone problem (due to shading of ASM):

```
[ERROR] Failures: 
[ERROR]   GradleTest.testJacksonBomDependency:36->build:64 FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':checkMetadata'.
> Dependencies of com.fasterxml.jackson.module:jackson-module-mrbean:2.15.1 are wrong in Gradle Metadata:
    POM:    com.fasterxml.jackson.core:jackson-core:2.15.1, com.fasterxml.jackson.core:jackson-databind:2.15.1, org.ow2.asm:asm:9.4
    Gradle: com.fasterxml.jackson.core:jackson-core:2.15.1, com.fasterxml.jackson.core:jackson-databind:2.15.1

```

**Comment by jjohannes** [comment]
Good to hear that the issue with `core` is fixed in 2.15.1 👍 

Regarding _mrbean_: was this just some timing issue with the releases and this will be fine in 2.16? Or is it unclear why this problem still exists?

**Comment by cowtowncoder** [comment]
@jjohannes I think it's a real problem, need to fix project definition similar to how Afterburner was changed. I forget what specifically is involved; or why it wasn't included as a failure earlier.
